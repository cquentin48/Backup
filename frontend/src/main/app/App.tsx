import React from "react";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import { ApolloProvider } from "@apollo/client";
import gqlClient from "./model/queries/client";
import "../res/css/App.css";
import ComputerPage from "./pages/computer/computerPage";
import TopBar from "./widget/topbar";

function App (): JSX.Element {
    return (
        <div className="App">
            <ApolloProvider client={gqlClient.get_query_client()}>
                <TopBar />
                <BrowserRouter>
                    <Routes>
                        <Route path="/" element={<p>Bonjour!</p>} />
                        <Route path="/test" element={<p>Element de test!</p>} />
                        <Route path="/computer/:id"
                            element={<ComputerPage graphqlQueryOperationManager={gqlClient} />} />
                    </Routes>
                </BrowserRouter>
            </ApolloProvider>
        </div>
    );
}

export default App;
