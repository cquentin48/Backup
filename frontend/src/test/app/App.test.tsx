import React from 'react';
import { render, screen } from '@testing-library/react';

import '@testing-library/jest-dom'
import App from '../../main/app/App';

describe("App root DOM node unit test suite", () => {
    test('renders learn react link', () => {
        // Acts
        const {asFragment} = render(<App/>);
        
        // Asserts
        expect(asFragment()).toMatchSnapshot()
    })
})
