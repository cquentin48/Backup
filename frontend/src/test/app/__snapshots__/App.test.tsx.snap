// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App root DOM node unit test suite renders learn react link 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      id="TopBar"
    >
      Top bar here!
    </div>
    <div>
      <p>
        Bonjour!
      </p>
      <p>
        <a
          href="/test"
        />
      </p>
    </div>
  </div>
</DocumentFragment>
`;
